src/components/Checkbox/Checkbox.tsx

```diff
@@ -12,20 +12,21 @@ import checkIcon from "../../assets/icons/check-icon";
 const classNames = require("classnames");
 
 const Label = styled.label`
-  &:not(.${styles.disabled}) {
+  :hover {
+    border-radius: ${(props) => (!props.isBorderRadius ? 0 : null)};
+  }
```

Comment:
if you have not an infinite number of variables better use classes.
Add modification class rounded.